app/models/manageiq/providers/ibm_power_hmc/infra_manager/provision_workflow.rb

```diff
@@ -9,11 +9,12 @@ def allowed_hosts_obj(_options = {})
 
     rails_logger('allowed_hosts_obj', 0)
     st = Time.zone.now
-    hosts_ids = load_ar_obj(src[:ems]).hosts.pluck(:id)
+    hosts_ids = load_ar_obj(src[:ems]).hosts.joins(:advanced_settings).where(:advanced_settings => {:name => "hmc_managed", :value => "true"}).pluck(:id)
```

Comment:
This would look a lot better if you added a scope to your host class:
```
scope :hmc_managed, -> { joins(:advanced_settings).where(:advanced_settings => {:name => "hmc_managed", :value => "true"}) }`
```
(untested)

then here you can do
```
load_ar_obj(src[:ems]).hosts.hmc_managed.pluck(:id)
```

Reply:
Seems that I would have to add the scope to the base `Host `class in order for this to work.

Reply:
Oh yeah that's not great, this is fine